### PR TITLE
Add support for graphs on GPUs and normalized graph laplacians

### PR DESCRIPTION
**Pull request by InfProbSciX**
This PR makes adds some functionality for symmetrically normalized graph laplacians and ensures that the tests work with tensors on GPUs. I also temporarily rewind `lab` and `plum-dispatch` versions to before the bug with lab (https://github.com/wesselb/lab/issues/6).

### CODE REVIEW

**Review by stoprightthere — COMMENTED** [review]
Looks good to me overall.

File: geometric_kernels/spaces/graph.py [hunk 1]
```diff
@@ -43,8 +45,12 @@ def _checks(adjacency):
     def dimension(self) -> int:
         return 0  # this is needed for the kernel math to work out
 
-    def set_laplacian(self, adjacency):
-        self._laplacian = degree(adjacency) - adjacency
+    def set_laplacian(self, adjacency, normalize_laplacian=False):
+        degree_matrix = degree(adjacency)
+        self._laplacian = degree_matrix - adjacency
+        if normalize_laplacian:
+            degree_inv_sqrt = B.linear_algebra.pinv(B.sqrt(degree_matrix))
```
[the remark this answers is not in the document]
Comment (InfProbSciX) [reply geometric_kernels/spaces/graph.py:52]: If the graph is disconnected, some of the degrees can be zero - `pinv` supports this case correctly (by having those elements by zero).

[the remark this answers is not in the document]
Comment (stoprightthere) [reply geometric_kernels/spaces/graph.py:52]: Hm, I think the `[None, :]`, `[:, None]` syntax should work hassle-free. Feel free to leave it as it is, if it breaks

[the remark this answers is not in the document]
Comment (InfProbSciX) [reply geometric_kernels/spaces/graph.py:52]: Thinking about this case some more - there are other spots where graphs with completely disconnected nodes aren't supported properly, lemme clean it up a bit.

[the remark this answers is not in the document]
Comment (InfProbSciX) [reply geometric_kernels/spaces/graph.py:52]: All done - I think that the multiplication issue can be a todo - most likely `pinv` would be more expensive than the matmul..!

[the remark this answers is not in the document]
Comment (stoprightthere) [reply geometric_kernels/spaces/graph.py:52]: Okay, makes sense. I'm approving this

**Review by stoprightthere — APPROVED** [review]
LGTM. Nice work!

**Pull request merged**